packages/fastify-renderer/src/node/renderers/react/ReactRenderer.tsx

```diff
@@ -204,14 +204,19 @@ export class ReactRenderer implements Renderer {
 
   private renderStreamingTemplate<Props>(app: JSX.Element, bus: RenderBus, ReactDOMServer: any, render: Render<Props>) {
     this.runHeadHooks(bus)
+    // Do nothing for now when using streaming template
+    bus.push('postRenderHead', null)
     const contentStream = ReactDOMServer.renderToNodeStream(app)
-    contentStream.on('end', () => this.runTailHooks(bus))
+    contentStream.on('end', () => {
+      this.runTailHooks(bus)
+    })
 
     return render.document({
       content: contentStream,
       head: bus.stack('head'),
       tail: bus.stack('tail'),
       props: render.props,
+      postRenderHead: bus.stack('postRenderHead'),
```

Comment:
Do you think it needs to be a different chunk of pushed elements on the render bus? I was thinking we could just push to the `head` stack from these hooks as well, and then the document template doesn't need to be any wiser / know about this thing at all

Reply:
That works just as good. I figure we'd likely have a separate stack if/when we handle streams and this would keep it consistent.